libs/core/src/schemas/events.schemas.ts

```diff
@@ -46,3 +46,52 @@ export const DiscordMessageCreated = z.object({
     z.literal('comment-create'),
   ]),
 });
+
+// All events should carry this common metadata
+export const EventMetadata = z.object({
+  createdAt: z.date().describe('When the event was emitted'),
+  // TODO: TBD
+  // aggregateType: z.enum(Aggregates).describe("Event emitter aggregate type")
+  // aggregateId: z.string().describe("Event emitter aggregate id")
+  // correlation: z.string().describe("Event correlation key")
+  // causation: z.object({}).describe("Event causation")
+});
+
+// on-chain contest manager events
+export const ContestManagerDeployed = EventMetadata.extend({
```

Comment:
I believe we should have two events for two different contests:

    - Recurring contest manager deployed: Namespace, interval, Contest manager address
    - Single contest manager deployed: Namespace, Length of the contest, Contest manager address

Reply:
@egetekiner is the length measured in days?